bindings/java/src/org/sleuthkit/datamodel/blackboardutils/ArtifactHelper.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Autopsy Forensic Browser
+ *
+ * Copyright 2019 Basis Technology Corp.
+ * Contact: carrier <at> sleuthkit <dot> org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.sleuthkit.datamodel.blackboardutils;
+
+import java.util.Collection;
+import org.apache.commons.lang3.StringUtils;
+import org.sleuthkit.datamodel.AbstractFile;
+import org.sleuthkit.datamodel.BlackboardAttribute;
+import org.sleuthkit.datamodel.SleuthkitCase;
+
+/**
+ * An abstract base class for classes that helps ingest modules create
+ * artifacts.
+ *
+ */
+class ArtifactHelper {
+
+	private final SleuthkitCase caseDb;
+	private final AbstractFile srcAbstractFile;	// artifact source
+	private final String moduleName;			// module creating the artifacts
+
+	/**
+	 * Creates an artifact helper.
+	 *
+	 * @param caseDb     Sleuthkit case db
+	 * @param moduleName name module using the helper
+	 * @param srcFile    source file
+	 */
+	protected ArtifactHelper(SleuthkitCase caseDb, String moduleName, AbstractFile srcFile) {
```

Comment:
Is it a requirement that this class should be subclassable from outside this package?  I assumed not. The reason I made the constructor protected was to prevent anyone outside of the package actually creating objects of this class( effectively making it an abstract base class).  I wanted this class to be abstract but Codacy (and you) complained about it not having any abstract method. I think a class should be allowed to be abstract without having any abstract method, depending on the use case. This class is good example of a class that should only be a base class but has no common method that all it's subclasses need to implement.

Reply:
I realize, belatedly, that 'protected' does not prevent a non-subclass in same package from accessing it anyway. So default/package it is.